zarr/v3/group.py

```diff
@@ -107,73 +116,276 @@ def from_json(
+        raise NotImplementedError
+
+    async def tree(self, expand=False, level=None) -> Any:
+        raise NotImplementedError
+
+    async def empty(self, **kwargs) -> AsyncArray:
+        raise NotImplementedError
+
+    async def zeros(self, **kwargs) -> AsyncArray:
+        raise NotImplementedError
+
+    async def ones(self, **kwargs) -> AsyncArray:
+        raise NotImplementedError
+
+    async def full(self, **kwargs) -> AsyncArray:
+        raise NotImplementedError
+
+    async def empty_like(self, prototype: AsyncArray, **kwargs) -> AsyncArray:
+        raise NotImplementedError
+
+    async def zeros_like(self, prototype: AsyncArray, **kwargs) -> AsyncArray:
+        raise NotImplementedError
+
+    async def ones_like(self, prototype: AsyncArray, **kwargs) -> AsyncArray:
+        raise NotImplementedError
+
+    async def full_like(self, prototype: AsyncArray, **kwargs) -> AsyncArray:
+        raise NotImplementedError
+
+    async def move(self, source: str, dest: str) -> None:
+        raise NotImplementedError
+
+
+@frozen
+class Group(SyncGroup, SyncMixin):
+    _async_group: AsyncGroup
+    _sync_configuration: field(factory=SyncConfiguration)
+
+    @classmethod
+    def create(
+        cls,
+        store: StoreLike,
+        *,
+        attributes: Optional[Dict[str, Any]] = None,
+        exists_ok: bool = False,
+        runtime_configuration: RuntimeConfiguration = RuntimeConfiguration(),
+    ) -> Group:
+        return cls._sync(
+            AsyncGroup.create,
+            store,
+            attributes=attributes,
+            exists_ok=exists_ok,
+            runtime_configuration=runtime_configuration,
+        )
+
+    @classmethod
+    def open(
+        cls,
+        store: StoreLike,
+        runtime_configuration: RuntimeConfiguration = RuntimeConfiguration(),
+    ) -> Group:
+        obj = cls._sync(AsyncGroup.open, store, runtime_configuration)
+        return cls(obj)
+
+    def __getitem__(self, path: str) -> Union[Array, Group]:
+        obj = self._sync(self._async_group.getitem, path)
+        if isinstance(obj, AsyncArray):
+            return Array(obj)
+        else:
+            return Group(obj)
+
+    def __delitem__(self, key):
+        raise NotImplementedError
+
+    def __iter__(self):
+        raise NotImplementedError
+
+    def __len__(self):
+        raise NotImplementedError
+
+    def __setitem__(self, key, value):
+        raise NotImplementedError
+
+    @property
+    def attrs(self):
+        return self._async_group.attrs
+
+    @property
+    def info(self):
+        return self._async_group.info
+
+    def update_attributes(self, new_attributes: Dict[str, Any]):
+        self._sync(self._async_group.update_attributes, new_attributes)
+        return self
+
+    @property
+    def nchildren(self) -> int:
+        return self._sync(self._async_group.nchildren)
+
+    @property
+    def children(self) -> List[Array, "Group"]:
```

Comment:
some exploratory work is needed to figure out how to make this (and similar methods) a proper iterator/generator instead of a list. The tricky bit is that the async method is returning an AsyncIterator and we need to synchronize that into a regular Iterator.